File: src/main/org/nlogo/prim/threed/_linkpitch.java
package org.nlogo.prim.threed;

import org.nlogo.agent.Link;
import org.nlogo.api.LogoException;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;
import org.nlogo.nvm.Syntax;

public final strictfp class _linkpitch extends Reporter
{
	@Override public Syntax syntax()
	{
		return Syntax.reporterSyntax
			( Syntax.TYPE_NUMBER , "---L" ) ;
	}
	@Override public Object report( Context context ) throws LogoException
	{
		return report_1( context ) ;
	}
	public double report_1( Context context )
		throws LogoException
	{
		try
		{
			Link link = (Link) context.agent ;
			return world.protractor().towardsPitch( link.end1() , link.end2() , true ) ;
		}
		catch ( org.nlogo.api.AgentException e )
		{
			throw new org.nlogo.nvm.EngineException
				( context , this , 
				  "there is no pitch of a link whose endpoints are in the same position" ) ;
		}
	}
}

File: src/main/org/nlogo/prim/threed/_linkpitch.scala
package org.nlogo.prim.threed

import org.nlogo.agent.Link
import org.nlogo.api.LogoException
import org.nlogo.nvm.{ Context, Reporter, Syntax }

class _linkpitch extends Reporter {
  override def syntax =
    Syntax.reporterSyntax(
      Syntax.TYPE_NUMBER, "---L")
  override def report(context: Context) = {
    val link = context.agent.asInstanceOf[Link]
    try Double.box(world.protractor.towardsPitch(link.end1, link.end2, true))
    catch {
      case e: org.nlogo.api.AgentException =>
        throw new org.nlogo.nvm.EngineException(
          context, this,
          "there is no pitch of a link whose endpoints are in the same position")
    }
  }
}
